pages/_document.tsx

```diff
@@ -1,13 +1,20 @@
-import { Html, Head, Main, NextScript } from "next/document"
+import Document, { Html, Head, Main, NextScript } from "next/document"
+import i18nextConfig from "../next-i18next.config"
 
-export default function Document() {
-  return (
-    <Html lang="en">
-      <Head />
-      <body>
-        <Main />
-        <NextScript />
-      </body>
-    </Html>
-  )
+class MyDocument extends Document {
+  render() {
+    const currentLocale = this.props.__NEXT_DATA__.query.locale || i18nextConfig.i18n.defaultLocale
+    const lang: string | undefined = Array.isArray(currentLocale) ? undefined : currentLocale;
```

Comment:
could we update this logic, so the current lang of the page is never undefined? 
a page lang should be always present in all cases

Reply:
you are right, but the lang logic was made, to satisfy the Html attribute type "lang" which can be string | undefined and the currentLocal type gives is string | string[].